flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableITCase.java

```diff
@@ -383,6 +383,147 @@ public void testKafkaSourceSinkWithMetadata() throws Exception {
 		deleteTestTopic(topic);
 	}
 
+	@Test
+	public void testKafkaSourceSinkWithKeyAndPartialValue() throws Exception {
+		if (isLegacyConnector) {
+			return;
+		}
+		// we always use a different topic name for each parameterized topic,
+		// in order to make sure the topic can be created.
+		final String topic = "key_partial_value_topic_" + format;
+		createTestTopic(topic, 1, 1);
+
+		// ---------- Produce an event time stream into Kafka -------------------
+		String groupId = standardProps.getProperty("group.id");
+		String bootstraps = standardProps.getProperty("bootstrap.servers");
+
+		// k_user_id and user_id have different data types to verify the correct mapping,
+		// fields are reordered on purpose
+		final String createTable = String.format(
+				"CREATE TABLE kafka (\n"
+						+ "  `k_user_id` BIGINT,\n"
+						+ "  `name` STRING,\n"
+						+ "  `timestamp` TIMESTAMP(3) METADATA,\n"
+						+ "  `k_event_id` BIGINT,\n"
+						+ "  `user_id` INT,\n"
+						+ "  `payload` STRING\n"
+						+ ") WITH (\n"
+						+ "  'connector' = 'kafka',\n"
+						+ "  'topic' = '%s',\n"
+						+ "  'properties.bootstrap.servers' = '%s',\n"
+						+ "  'properties.group.id' = '%s',\n"
+						+ "  'scan.startup.mode' = 'earliest-offset',\n"
+						+ "  'key.format' = '%s',\n"
+						+ "  'key.fields' = 'k_event_id; k_user_id',\n"
+						+ "  'key.fields-prefix' = 'k_',\n"
+						+ "  'value.format' = '%s',\n"
+						+ "  'value.fields-include' = 'EXCEPT_KEY'\n"
+						+ ")",
+				topic,
+				bootstraps,
+				groupId,
+				format,
+				format);
+
+		tEnv.executeSql(createTable);
+
+		String initialValues = "INSERT INTO kafka\n"
+				+ "VALUES\n"
+				+ " (1, 'name 1', TIMESTAMP '2020-03-08 13:12:11.123', 100, 1, 'payload 1'),\n"
+				+ " (2, 'name 2', TIMESTAMP '2020-03-09 13:12:11.123', 101, 2, 'payload 2'),\n"
+				+ " (3, 'name 3', TIMESTAMP '2020-03-10 13:12:11.123', 102, 3, 'payload 3')";
```

Comment:
minor: how about having differnet values for `user_id`?  This can help to verify we can extract correct values when name clashes.